### PR TITLE
chore: migrate libuild code and test

### PR DESCRIPTION
**Pull request by 10Derozan**
### <samp>🤖 Generated by Copilot at f0c11e0</samp>

This pull request adds a new package `libuild-core` that provides a tool for building modern JavaScript libraries using esbuild and other plugins. It includes the source code, tests, configuration, documentation, and scripts for the package. It also updates the `.eslintignore` file to ignore the integration tests directory.

### <samp>🤖 Generated by Copilot at f0c11e0</samp>

*  Add `libuild-core` package as a tool for building modern JavaScript libraries using esbuild and other plugins (F1-F10, F15-F18)

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/libuild/libuild-core/.mocharc.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const path = require('path');
```
Comment (chenjiahan) [inline packages/libuild/libuild-core/.mocharc.js:1]: We should migrate from mocha to vitest later.

File: packages/libuild/libuild-core/LICENSE.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
+# Libuild license
```
Comment (chenjiahan) [inline packages/libuild/libuild-core/LICENSE.md:1]: Please use the License of Modern.js

File: pnpm-lock.yaml [hunk 3]
```diff
@@ -3254,11 +3254,485 @@ importers:
         specifier: ^5
         version: 5.0.4
 
+  packages/libuild/libuild-core:
```
Comment (chenjiahan) [inline pnpm-lock.yaml:3257]: The lock file size has increased by about 3000+ lines, we should optimize it in the future.

File: packages/libuild/libuild-core/.mocharc.js (hunk 1 above)
Comment (10Derozan) [reply packages/libuild/libuild-core/.mocharc.js:1]: It won't be included in this PR.

File: packages/libuild/libuild-core/CHANGELOG.json [hunk 4]
```diff
@@ -0,0 +1,610 @@
+{
```
[the remark this answers is not in the document]
Comment (10Derozan) [reply packages/libuild/libuild-core/CHANGELOG.json:1]: I'll remove them.

File: pnpm-lock.yaml (hunk 3 above)
Comment (10Derozan) [reply pnpm-lock.yaml:3257]: Approve it

File: packages/libuild/libuild-core/LICENSE.md (hunk 2 above)
Comment (10Derozan) [reply packages/libuild/libuild-core/LICENSE.md:1]: Get it.

File: packages/libuild/libuild-core/CHANGELOG.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (10Derozan) [reply packages/libuild/libuild-core/CHANGELOG.json:1]: Should we remove the changelog information generated by rush?

[the remark this answers is not in the document]
Comment (chenjiahan) [reply packages/libuild/libuild-core/CHANGELOG.json:1]: I prefer remove them because their format is not consistent with the changeset generated changelog

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**